### PR TITLE
INTMDB-426: Alert Configuration -- Api Token erroneous changes

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Alert Configuration -- Api Token erroneous changes adapt logic to ignore API returning obfuscated values for API keys

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_alert_configuration.go [hunk 1]
```diff
@@ -731,33 +731,38 @@ func expandAlertConfigurationNotification(d *schema.ResourceData) ([]matlas.Noti
 	return notifications, nil
 }
 
-func flattenAlertConfigurationNotifications(notifications []matlas.Notification) []map[string]interface{} {
+func flattenAlertConfigurationNotifications(d *schema.ResourceData, notifications []matlas.Notification) []map[string]interface{} {
+	notificationsSchema, err := expandAlertConfigurationNotification(d)
+	if err != nil {
+		return nil
+	}
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_alert_configuration.go:738]: [q] Can you add a comment with the reason why we need this new schema? Thanks!

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: mongodbatlas/resource_mongodbatlas_alert_configuration.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_alert_configuration.go:738]: @andreaangiolillo I pull in existing values set in TF file to set API keys in state file to ignore values that come back from API side that are masked

**Review by Zuhairahmed — APPROVED** [review]
Thanks @martinstibbe. There are a few more comments in GitHub Issue. Can you double check this fix addresses those as well? Otherwise LGTM. 
https://github.com/mongodb/terraform-provider-mongodbatlas/issues/863

**Pull request merged**